GenericsHomework/Node.cs

```diff
@@ -0,0 +1,68 @@
+namespace GenericsHomework;
+public class Node<T>
+{
+    // Value of the node
+    public T Value { get; set; }
+
+    // Pointer to the next node
+    public Node<T> Next { get; private set; }
+
+    // Constructor that takes a value
+    public Node(T value)
+    {
+        Value = value;
+       
+        Next = this; // By default, the Next pointer refers back to itself
+     
+      
+    }
+
+    public override string ToString()
+    {
+        return Value!.ToString()!;

```

Comment:
```suggestion
    public override string? ToString()
    {
        if (Value == null) return null;
        return Value.ToString()!;

```
Close! What if Value is null?